File: 0009-palindrome-number/0009-palindrome-number.py
class Solution(object):
    def isPalindrome(self, x):
        """
        :type x: int
        :rtype: bool
        """
        if x < 0:
            return False

        reverse = 0
        temp = x

        while temp != 0:
            num = temp % 10
            reverse = reverse * 10 + num
            temp //= 10

        return reverse == x

File: 0009-palindrome-number/0009-palindrome-number.java
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0  || x != 0 && x%10 == 0) {
            return false;
        }

        int test = 0;
        while (x > test) {
            test = test*10 + x%10;
            x/=10;
        }

        return (x == test || x == test/10);
    }
}